### PR TITLE
[AG-609] pool list updater for algebrav1

### PR DESCRIPTION
**Pull request by it4rb**

### CODE REVIEW

File: pkg/source/algebrav1/queries.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		lastPoolIdsQ,
+	}
+
+	// Add subgraphError: allow
+	t, err := template.New("poolsListQuery").Parse(`{
+		pools(
+			{{ if .AllowSubgraphError }}subgraphError: allow,{{ end }}
+			where: {
+				createdAtTimestamp_gte: {{ .LastCreatedAtTimestamp }}
+				{{ .LastPoolIdsQuery }}
```
Comment (lehainam-dev) [inline pkg/source/algebrav1/queries.go:47]: Does it should be `.LastPoolIdsQ` instead?
```suggestion
				{{ .LastPoolIdsQ }}
```

File: pkg/source/algebrav1/config.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package algebrav1
+
+type Config struct {
+	DexID              string
+	SubgraphAPI        string `json:"subgraphAPI"`
+	AllowSubgraphError bool   `json:"allowSubgraphError"`
```
Comment (piavgh) [inline pkg/source/algebrav1/config.go:6]: I think we don't need this

File: pkg/source/algebrav1/errors.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"errors"
+)
+
+var (
+	ErrUnmarshalVolLiq     = errors.New("failed to unmarshal volumePerLiquidityInBlock")
+	ErrMaxBinarySearchLoop = errors.New("max binary search loop reached")
+	ErrStaleTimepoints     = errors.New("getting stale timepoint data")
+	ErrTickNil             = errors.New("tick is nil")
+	ErrV3TicksEmpty        = errors.New("v3Ticks empty")
```
Comment (piavgh) [inline pkg/source/algebrav1/errors.go:12]: v3 ticks might not relate to Algebra V1, I think we should update the naming here

File: pkg/source/algebrav1/pool_list_updater.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	graphqlClient := graphqlPkg.NewWithTimeout(cfg.SubgraphAPI, graphQLRequestTimeout)
+
+	return &PoolsListUpdater{
+		config:        cfg,
+		graphqlClient: graphqlClient,
+	}
+}
+
+func (d *PoolsListUpdater) getPoolsList(ctx context.Context, lastCreatedAtTimestamp *big.Int, lastPoolIds []string, first, skip int) ([]SubgraphPool, error) {
+	allowSubgraphError := d.config.AllowSubgraphError
```
Comment (piavgh) [inline pkg/source/algebrav1/pool_list_updater.go:36]: this workaround is not necessary

File: pkg/source/handler.go [hunk 5]
```diff
@@ -2,10 +2,12 @@ package source
 
 import (
 	"fmt"
+
 	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/source/pearl"
```
Comment (piavgh) [inline pkg/source/handler.go:6]: please help to re-organize this import

File: pkg/source/algebrav1/queries.go (hunk 1 above)
Comment (piavgh) [inline pkg/source/algebrav1/queries.go:47]: It's better to add a comment to explain the purpose of `LastPoolIdsQuery` in this function

**Review by piavgh — APPROVED** [review]

File: pkg/source/algebrav1/queries.go (hunk 1 above)
Comment (it4rb) [reply pkg/source/algebrav1/queries.go:48]: added above

File: pkg/source/handler.go (hunk 5 above)
Comment (it4rb) [reply pkg/source/handler.go:6]: um this is already formatted

**Pull request merged**